sdks/java/io/solace/src/main/java/org/apache/beam/sdk/io/solace/data/Solace.java

```diff
@@ -68,4 +68,307 @@ public String getName() {
       return name;
     }
   }
+  /** Represents a Solace destination type. */
+  public enum DestinationType {
+    TOPIC,
+    QUEUE,
+    UNKNOWN
+  }
+
+  /** Represents a Solace message destination (either a Topic or a Queue). */
+  @AutoValue
+  @DefaultSchema(AutoValueSchema.class)
+  public abstract static class Destination {
+    /**
+     * Gets the name of the destination.
+     *
+     * @return The destination name.
+     */
+    @SchemaFieldNumber("0")
+    public abstract String getName();
+
+    /**
+     * Gets the type of the destination (TOPIC, QUEUE or UNKNOWN).
+     *
+     * @return The destination type.
+     */
+    @SchemaFieldNumber("1")
+    public abstract DestinationType getType();
+
+    static Builder builder() {
+      return new AutoValue_Solace_Destination.Builder();
+    }
+
+    @AutoValue.Builder
+    abstract static class Builder {
+      abstract Builder setName(String name);
+
+      abstract Builder setType(DestinationType type);
+
+      abstract Destination build();
+    }
+  }
+
+  /** Represents a Solace message record with its associated metadata. */
+  @AutoValue
+  @DefaultSchema(AutoValueSchema.class)
+  public abstract static class Record {
+    /**
+     * Gets the unique identifier of the message, a string for an application-specific message
+     * identifier.
+     *
+     * <p>Mapped from {@link BytesXMLMessage#getApplicationMessageId()}
+     *
+     * @return The message ID, or null if not available.
+     */
+    @SchemaFieldNumber("0")
+    public abstract @Nullable String getMessageId();
```

Comment:
What would Solace respond with if the message ID was an empty string instead of null?

Reply:
I'm not sure, [the documentation for this method](https://docs.solace.com/API-Developer-Online-Ref-Documentation/java/com/solacesystems/jcsmp/XMLMessage.html#getApplicationMessageId--) doesn't specify this. But since this is a read connector, and a user will not create this class manually, should we worry about this?